### DISCUSSION TITLE
error 500 Datasource is not configured fully

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2659] [Help]
Hi there.

I'm trying to build a details/update page for a custom postgresql sql table. I'm copying from an automatic details screen created by bb.

So I've created the same structure as the page built by bb:

```
Url: /test/:id
>Data provider with filter id = {{field from query}}
  >Repeater
     >Form
        >Field Group
```

If I create an input field there's no fields to pick from the query (the field dropdown is empty).
If I try to auto fill the form nothing happens.
If I open the page I get this error - pasting below.

Any idea what could be the problem?

Thanks.

`ERROR [1632048548479] (29 on 889e31031584): Datasource is not configured fully.
    req: {
      "id": 5277,
      "method": "GET",
      "url": "/api/tables/query_datasource_plus_17ef5a0ba9354318a14c6b75362d1d52_f5419e7c36064077912aabaa737f6c27",
      "headers": {
        "host": "localhost:10000",
        "upgrade-insecure-requests": "1",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.63 Safari/537.36",
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
        "sec-gpc": "1",
        "sec-fetch-site": "none",
        "sec-fetch-mode": "navigate",
        "sec-fetch-user": "?1",
        "sec-fetch-dest": "document",
Trace: Datasource is not configured fully.
    at /app/dist/api/index.js:67:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
        "accept-encoding": "gzip, deflate, br",
        "accept-language": "pt-PT,pt;q=0.9,en-US;q=0.8,en;q=0.7",
        "cookie": "budibase:currentapp=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhcHBJZCI6ImFwcF9kZXZfZTM4ZDVhNmEwMjFmNGUxODg5MDE2MjcyODg1ZWFmNWYiLCJpYXQiOjE2MzE4MDQwNzF9.AqexmeAFkp3Y0qI-ONj8Y0kGrNqR2t_lRfHVkwma2m0; budibase:auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJ1c19mZDliMTA5M2EzMWQ0OGE3YTAzNWQwYThkYjcwNjE3NCIsInNlc3Npb25JZCI6IjA4MjQ2MjI1OTc2ODRkZTZiNDcyNTBmNGY4Y2ZiNmY2IiwidGVuYW50SWQiOiJkZWZhdWx0IiwiaWF0IjoxNjMxOTkxODk5fQ.MxqzGTcbG7wZQjj5S-UzknPHipQiSXGxyRUVwQc4xy0; budibase:auth.sig=RFJ5RDp6prl7P1uZhfccvQSBhd0",
        "x-forwarded-proto": "http",
        "x-request-id": "389b7f65-2bbd-43ea-a9f7-f2fd97820461",
        "x-envoy-expected-rq-timeout-ms": "15000",
        "content-length": "0"
      },
      "remoteAddress": "::ffff:172.18.0.5",
      "remotePort": 34450
    }
INFO  [1632048548480] (29 on 889e31031584): request errored
    res: {
      "statusCode": 500,
      "headers": {
        "vary": "Accept-Encoding",
        "content-type": "application/json; charset=utf-8",
        "content-length": "14"
      }
    }
    err: {
      "type": "Error",
      "message": "failed with status code 500",
      "stack":
          Error: failed with status code 500
              at ServerResponse.onResFinished (/app/node_modules/pino-http/logger.js:74:38)
              at ServerResponse.emit (events.js:326:22)
              at ServerResponse.EventEmitter.emit (domain.js:483:12)
              at ServerResponse.emitted (/app/node_modules/emitter-listener/listener.js:134:21)
              at onFinish (_http_outgoing.js:735:10)
              at onCorkedFinish (_stream_writable.js:673:5)
              at afterWrite (_stream_writable.js:490:5)
              at afterWriteTick (_stream_writable.js:477:10)
              at processTicksAndRejections (internal/process/task_queues.js:83:21)
    }
    responseTime: 186
    req: {
      "id": 5277,
      "method": "GET",
      "url": "/api/tables/query_datasource_plus_17ef5a0ba9354318a14c6b75362d1d52_f5419e7c36064077912aabaa737f6c27",
      "headers": {
        "host": "localhost:10000",
        "upgrade-insecure-requests": "1",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.63 Safari/537.36",
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
        "sec-gpc": "1",
        "sec-fetch-site": "none",
        "sec-fetch-mode": "navigate",
        "sec-fetch-user": "?1",
        "sec-fetch-dest": "document",
        "accept-encoding": "gzip, deflate, br",
        "accept-language": "pt-PT,pt;q=0.9,en-US;q=0.8,en;q=0.7",
        "cookie": "budibase:currentapp=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhcHBJZCI6ImFwcF9kZXZfZTM4ZDVhNmEwMjFmNGUxODg5MDE2MjcyODg1ZWFmNWYiLCJpYXQiOjE2MzE4MDQwNzF9.AqexmeAFkp3Y0qI-ONj8Y0kGrNqR2t_lRfHVkwma2m0; budibase:auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJ1c19mZDliMTA5M2EzMWQ0OGE3YTAzNWQwYThkYjcwNjE3NCIsInNlc3Npb25JZCI6IjA4MjQ2MjI1OTc2ODRkZTZiNDcyNTBmNGY4Y2ZiNmY2IiwidGVuYW50SWQiOiJkZWZhdWx0IiwiaWF0IjoxNjMxOTkxODk5fQ.MxqzGTcbG7wZQjj5S-UzknPHipQiSXGxyRUVwQc4xy0; budibase:auth.sig=RFJ5RDp6prl7P1uZhfccvQSBhd0",
        "x-forwarded-proto": "http",
        "x-request-id": "389b7f65-2bbd-43ea-a9f7-f2fd97820461",
        "x-envoy-expected-rq-timeout-ms": "15000",
        "content-length": "0"
      },
      "remoteAddress": "::ffff:172.18.0.5",
      "remotePort": 34450
    }`

**Comment by shogunpurple** [comment]
@dncpax it appears that your datasource has been misconfigured somehow.

Here's the code that throws that error - looks like your datasource has no entities, which is odd.
```
  if (!datasource || !datasource.entities) {
    throw "Datasource is not configured fully."
  }
  ```
  
  Did it work with your autoscreens before, and only started happening when you created your own screen from scratch?

**Reply by dncpax** [reply]
this is really odd... I use a sql query to show a list. Then I created a button to open a details screen for an item on that list. Then I created this screen by mimicking a screen bb created for another table that is on the public schema. The reason for this approach is because my table is not on the public schema so I had to create a sql query.
Now on the details screen I'm able to get the form working by manually writing the fields names in each text/date/number field I create. This works. I can still see the 500 error in the browser console but nothing's visibly wrong in the app.
